Update build system docs (#4762)
woocommerce/woocommerce-gutenberg-products-block: docs/contributors/js-build-system.md

## Babel

> Almost all our code is transpiled by Babel, this allows us to use the latest JavaScript technologies without having to worry about browser support.

Reviewer: ```suggestion
Most of our code is transpiled by Babel. This allows us to use the latest JavaScript technologies without affecting browser support.
```
